metrics/collectors/counter.lua

```diff
@@ -9,4 +9,8 @@ function Counter:inc(num, label_pairs)
     Shared.inc(self, num, label_pairs)
 end
 
+function Counter:reset(label_pairs)
```

Comment:
The original request was about counter initialization. Initialization with `counter:reset(label_pairs)` seems kinda weird. How about adding an alias for `reset` as a little syntactic sugar?

```lua
Counter.initialize = reset
```

Reply:
init / reinit / reset, who cares. Reset is better than inc(0), and that's the purpose.

Reply:
Init which resets the value may be considered weird too.